modules/registrars/openprovider/Controllers/System/RenewDomainController.php

```diff
@@ -81,17 +77,18 @@ public function renew($params)
         // We did not have a true isInRedemptionGracePeriod or isInGracePeriod. Fall back on the legacy code
         // for older WHMCS versions.
 
-        try
-        {
-            if(!$domainOp['softQuarantineExpiryDate']) {
-                $this->apiHelper->renewDomain($domainOp['id'], $period);
-            } elseif ((new Carbon($domainOp['softQuarantineExpiryDate'], 'Europe/Amsterdam'))->gt(Carbon::now('Europe/Amsterdam'))) {
-                $this->apiHelper->restoreDomain($domainOp['id']);
+        try {
+            if ($domainOp['status'] == 'DEL') {
+                if ($domainOp['softQuarantineExpiryDate'] && (new Carbon($domainOp['softQuarantineExpiryDate'], 'Europe/Amsterdam'))->gt(Carbon::now('Europe/Amsterdam'))) {
+                    $this->apiHelper->restoreDomain($domainOp['id']);
+                } elseif ($domainOp['hardQuarantineExpiryDate'] && (new Carbon($domainOp['hardQuarantineExpiryDate'], 'Europe/Amsterdam'))->gt(Carbon::now('Europe/Amsterdam'))) {
+                    return ['error' => "Domain is past the grace period and additional costs may be applied. Please check the domain in your reseller control panel for more information"];
+                } else {
+                    return ['error' => "Domain has been deleted Please check the domain in your reseller control panel"];
+                }
             } else {
-                // This only happens when the isInRedemptionGracePeriod was not true.
-                throw new Exception("Domain has expired and additional costs may be applied. Please check the domain in your reseller control panel", 1);
+                $this->apiHelper->renewDomain($domainOp['id'], $period);
```

Comment:
don't we need to check other statuses?

Reply:
Restore domain API or specific errors from WHMCS side need to be shown only for DEL state domain from WHMCS side and
for all other statuses like "Active", "Req", "Pending" e.t.c, It is enough to show the results thrown by OP side after calling renew APICommand.